code/game/objects/effects/decals/crayon.dm

```diff
@@ -1,34 +1,61 @@
 /obj/effect/decal/cleanable/crayon
-	name = "rune"
-	desc = "A rune drawn in crayon."
+	name = "generic scribble"
+	desc = "A generic scribble.  You probably shouldn't see this."
 	icon = 'icons/obj/rune.dmi'
 	layer = 2.1
-	anchored = 1
+	anchored = TRUE
 
-/obj/effect/decal/cleanable/crayon/Initialize(mapload, main = "#FFFFFF", shade = "#000000", var/type = "rune")
-	. = ..()
+/obj/effect/decal/cleanable/crayon/line/
+	name = "crayon marking"
+	desc = "A marking drawn in crayon."
+	icon = 'icons/obj/smooth/crayonline-smooth.dmi'
+	icon_state = "preview"
+	layer = 2.1
+	anchored = TRUE
+	smooth = SMOOTH_TRUE
 
-	name = type
-	desc = "A [type] drawn in crayon."
+/obj/effect/decal/cleanable/crayon/line/Initialize(mapload, main = "#FFFFFF", shade = "#000000", var/drawtype = "line", var/utensiltype = "crayon")
+	. = ..(mapload, main, shade, drawtype, utensiltype)
+	if (mapload)
+		queue_smooth(src)
+	else
+		smooth_icon(src)
+		for (var/obj/effect/decal/cleanable/crayon/line/C in orange(1, src))
+			smooth_icon(C)
 
-	switch(type)
+/obj/effect/decal/cleanable/crayon/Initialize(mapload, main = "#FFFFFF", shade = "#000000", var/drawtype = "rune", var/utensiltype = "crayon")
+	. = ..()
+	if(drawtype == "line")
+		if(utensiltype == "chalk")
+			desc = "A marking drawn in chalk."
+			icon = 'icons/obj/smooth/chalkline-smooth.dmi'
+		add_hiddenprint(usr)
+		color = main
+		return
+	if(utensiltype == "chalk")
+		smooth = SMOOTH_TRUE			
+	name = drawtype
+	desc = "A [drawtype] drawn in [utensiltype]."
+	switch(drawtype)
 		if("rune")
-			type = "rune[rand(1,6)]"
+			drawtype = "rune[rand(1,6)]"
+			desc = "A strange collection of symbols drawn in [utensiltype]."
 		if("graffiti")
-			type = pick("amyjon","face","matt","revolution","engie","guy","end","dwarf","uboa")
-
-	var/icon/mainOverlay = SSicon_cache.crayon_cache[type]
+			drawtype = pick("amyjon","face","matt","revolution","engie","guy","end","dwarf","uboa")
+	var/icon/mainOverlay = SSicon_cache.crayon_cache[drawtype]
 	if (!mainOverlay)
-		mainOverlay = new/icon('icons/effects/crayondecal.dmi',"[type]",2.1)
+		mainOverlay = new/icon('icons/effects/crayondecal.dmi',"[drawtype]",2.1)
 		mainOverlay.Blend(main,ICON_ADD)
-		SSicon_cache.crayon_cache[type] = mainOverlay
+		SSicon_cache.crayon_cache[drawtype] = mainOverlay
 
-	var/icon/shadeOverlay = SSicon_cache.crayon_cache["[type]_s"]
+	var/icon/shadeOverlay = SSicon_cache.crayon_cache["[drawtype]_s"]
 	if (!shadeOverlay)
-		shadeOverlay = new/icon('icons/effects/crayondecal.dmi',"[type]s",2.1)
+		shadeOverlay = new/icon('icons/effects/crayondecal.dmi',"[drawtype]s",2.1)
 		shadeOverlay.Blend(shade,ICON_ADD)
-		SSicon_cache.crayon_cache["[type]_s"] = shadeOverlay
+		SSicon_cache.crayon_cache["[drawtype]_s"] = shadeOverlay
 
 	add_overlay(list(mainOverlay, shadeOverlay))
 
-	add_hiddenprint(usr)
+	SSicon_cache.crayon_cache.Cut()
```

Comment:
why are you cutting the cache here?

Reply:
I don't completely understand SSicon_cache, but from what I could puzzle out anyways, what would happen is the crayon code as it was originally would store the mainOverlay and the shadeOverlay in the cache, then never get rid of them.  The cache sticks around between drawings, so you'd wind up with all the previous drawings you'd drawn with the crayon in the cache.

The practical effect of this was that if you went to go draw something with your crayon, sometimes instead of getting what you chose (a rune, an arrow, a letter, etc) you'd get something you'd already drawn before... even if it was with a different crayon!  (Or chalk!)  So, not understand entirely what was going on, I tried out cutting the cache, and it fixed the problem.

Reply:
@NonQueueingMatt Did you have any other questions about this?

Reply:
You're adding to the icon cache in lines 88 and 82, but then you're also completely wiping the cache (that is what Cut() does), why?

Reply:
Alright, now I understand caches a bit better.  It would add a cache listing for that drawtype, but fail to distinguish between colors, so whenever a drawtype was retrieved from the cache, it would be the color of the first crayon/chalk that drew it.  Made it use color now.